File: japiwdbe/WdbeQCmdList.java
/**
	* \file WdbeQCmdList.h
	* Java API code for record of table TblWdbeQCmdList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQCmdList {
	
	public WdbeQCmdList(
				int jnum
				, String sref
				, String Fullsref
				, String srefRefIxVTbl
				, String titRefIxVTbl
				, String stubRefUref
				, String srefIxVRettype
				, String titIxVRettype
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Fullsref = Fullsref;
		this.srefRefIxVTbl = srefRefIxVTbl;
		this.titRefIxVTbl = titRefIxVTbl;
		this.stubRefUref = stubRefUref;
		this.srefIxVRettype = srefIxVRettype;
		this.titIxVRettype = titIxVRettype;
	};

	public int jnum;
	public String sref;
	public String Fullsref;
	public String srefRefIxVTbl;
	public String titRefIxVTbl;
	public String stubRefUref;
	public String srefIxVRettype;
	public String titIxVRettype;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQCmdList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Fullsref = Xmlio.extractStringUclc(doc, basexpath, "Fullsref", "fsr", null, 0);
			srefRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefRefIxVTbl", "ret", null, 0);
			titRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titRefIxVTbl", "ret2", null, 0);
			stubRefUref = Xmlio.extractStringUclc(doc, basexpath, "stubRefUref", "reu", null, 0);
			srefIxVRettype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVRettype", "rty", null, 0);
			titIxVRettype = Xmlio.extractStringUclc(doc, basexpath, "titIxVRettype", "rty2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQCmdList.cpp
/**
	* \file WdbeQCmdList.cpp
	* API code for table TblWdbeQCmdList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQCmdList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQCmdList
 ******************************************************************************/

WdbeQCmdList::WdbeQCmdList(
			const uint jnum
			, const string sref
			, const string Fullsref
			, const string srefRefIxVTbl
			, const string titRefIxVTbl
			, const string stubRefUref
			, const string srefIxVRettype
			, const string titIxVRettype
		) :
			jnum(jnum)
			, sref(sref)
			, Fullsref(Fullsref)
			, srefRefIxVTbl(srefRefIxVTbl)
			, titRefIxVTbl(titRefIxVTbl)
			, stubRefUref(stubRefUref)
			, srefIxVRettype(srefIxVRettype)
			, titIxVRettype(titIxVRettype)
		{
};

bool WdbeQCmdList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQCmdList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Fullsref", "fsr", Fullsref);
		extractStringUclc(docctx, basexpath, "srefRefIxVTbl", "ret", srefRefIxVTbl);
		extractStringUclc(docctx, basexpath, "titRefIxVTbl", "ret2", titRefIxVTbl);
		extractStringUclc(docctx, basexpath, "stubRefUref", "reu", stubRefUref);
		extractStringUclc(docctx, basexpath, "srefIxVRettype", "rty", srefIxVRettype);
		extractStringUclc(docctx, basexpath, "titIxVRettype", "rty2", titIxVRettype);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQCmdList
 ******************************************************************************/

ListWdbeQCmdList::ListWdbeQCmdList() {
};

ListWdbeQCmdList::ListWdbeQCmdList(
			const ListWdbeQCmdList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCmdList(*(src.nodes[i]));
};

ListWdbeQCmdList& ListWdbeQCmdList::operator=(
			const ListWdbeQCmdList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCmdList(*(src.nodes[i]));

	return *this;
};

ListWdbeQCmdList::~ListWdbeQCmdList() {
	clear();
};

void ListWdbeQCmdList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQCmdList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQCmdList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQCmdList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQCmdList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQCmdList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQCmdList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
